### PR TITLE
feat(bpmn): reload element templates on file save

### LINKED ISSUE #3471: Local element templates not recognized after initially saving a file
### Describe the bug

The modeler resolves element templates locally to a saved file. When the file is saved initially to a location we don't re-load the templates (in accordance to the new location). I have to forcefully reload the app to make to recognize the template.

```plain
.camunda
  element-templates
    some-template.json
myDiagram.bpmn <-- NEW FILE, should recognize <some-template> once saved
```

### Steps to reproduce

1. Create a local element template (`$CWD/.camunda/element-templates/some-template.json`)
2. Create new diagram, save to `$CWD`
3. See that template is not available

### Expected behavior

Once we safe a diagram (or location changed) _then_ locally available element templates are reloaded.

### Environment

- OS: Linux (any)
- Camunda Modeler Version: v5.8.0
- Execution Platform: any
- Installed plug-ins: none

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by marstamm**
Recording 2023-07-28 at 11 23 20

closes #3471

### CODE REVIEW

File: client/src/app/tabs/MultiSheetTab.js [hunk 1]
```diff
@@ -388,6 +388,7 @@ export class MultiSheetTab extends CachedComponent {
         <TabContainer className="content tab">
           <Editor
             ref={ this.editorRef }
+            file={ tab.file }
```
Comment (philippfromme) [inline client/src/app/tabs/MultiSheetTab.js:391]: I was wondering whether we could simply detect the change from new to not new (`isNew`) so we don't have to pass the entire file to the editor?

Comment (marstamm) [reply client/src/app/tabs/MultiSheetTab.js:391]: If we already have the file saved and change the location via `save file as`, we change the location and the potential templates without changes to `isNew`

Comment (philippfromme) [reply client/src/app/tabs/MultiSheetTab.js:391]: So we're only interested in `path` changes, right?

Comment (marstamm) [reply client/src/app/tabs/MultiSheetTab.js:391]: I decided to use the `file` Object and not just the path to check for any save, not just those that change the file path. This allows us to reload the element templates when the bpmn stays in the same place but element templates were added/modified externally.

We can discuss if this is expected behavior

**Review by philippfromme — APPROVED** [review]
✅

**Pull request merged**